src/common/entities/unique_codes.entity.ts

```diff
@@ -0,0 +1,26 @@
+import { Entity, PrimaryKey, Property, OneToOne, EntityRepositoryType } from "@mikro-orm/core";
+
+import { UniqueCodeRepository } from "@/unique-code/unique-code.repository";
+
+import { CustomBaseEntity } from "./custom-base.entity";
+import { Teacher } from "./teachers.entity";
+
+@Entity({ tableName: "unique_codes", repository: () => UniqueCodeRepository })
+export class UniqueCode extends CustomBaseEntity {
+  [EntityRepositoryType]?: UniqueCodeRepository;
+
+  @PrimaryKey()
+  id!: number;
+
+  @Property({ unique: true })
+  code!: string;
+
+  @Property()
+  email!: string;
+
+  @Property({ default: 0 })
+  usageCount!: number;
+
+  @OneToOne(() => Teacher, (teacher) => teacher.uniqueCode, { nullable: true })
+  teacher?: Teacher;
```

Comment:
We should remove this relationship. There would be no record in the teachers table prior to registering and once registered, there is no use case for the unique code.